core/sql/executor/ExHbaseAccess.cpp

```diff
@@ -2808,53 +2808,51 @@ short ExHbaseAccessTcb::setupHbaseFilterPreds()
       (hbaseAccessTdb().listOfHbaseFilterColNames()->numEntries() == 0))
     return 0;
 
-  if (! hbaseFilterValExpr())
-    return 0;
-
-  ex_queue_entry *pentry_down = qparent_.down->getHeadEntry();
-
-  workAtp_->getTupp(hbaseAccessTdb().hbaseFilterValTuppIndex_)
-    .setDataPointer(hbaseFilterValRow_);
-  
-  ex_expr::exp_return_type evalRetCode =
-    hbaseFilterValExpr()->eval(pentry_down->getAtp(), workAtp_);
-  if (evalRetCode == ex_expr::EXPR_ERROR)
-    {
-      return -1;
-    }
-
-  ExpTupleDesc * hfrTD =
-    hbaseAccessTdb().workCriDesc_->getTupleDescriptor
-    (hbaseAccessTdb().hbaseFilterValTuppIndex_);
-  
-  hbaseFilterValues_.clear();
-  for (Lng32 i = 0; i <  hfrTD->numAttrs(); i++)
-    {
-      Attributes * attr = hfrTD->getAttr(i);
-  
-      if (attr)
-	{
-	  NAString value(getHeap());
-	  if (attr->getNullFlag())
-	    {
-	      char nullValChar = 0;
-
-	      short nullVal = *(short*)&hbaseFilterValRow_[attr->getNullIndOffset()];
-
-	      if (nullVal)
-		nullValChar = -1;
-	      value.append((char*)&nullValChar, sizeof(char));
-	    }	  
-
-	  char * colVal = &hbaseFilterValRow_[attr->getOffset()];
-
-	  value.append(colVal,
-		       attr->getLength(&hbaseFilterValRow_[attr->getVCLenIndOffset()]));
-
-	  hbaseFilterValues_.insert(value);
-	}
-    }
-
+  if (hbaseFilterValExpr()){// with pushdown V2 it can be null if we have only unary operation
```

Comment:
Yes, this is a problem with tab settings. When using your favorite editor, it's very important to set it to use 8 spaces for a tab, otherwise it will mess up the indentation for everyone else (and the indentation of many existing files will be wrong). The editor should also be set to insert spaces, not tabs, into the source file. Please do not check in any files with the wrong tab setting.